pkg/kube/kube_handler.go

```diff
@@ -649,36 +649,49 @@ func filterAllowedNamespaces(userClientset combinedClientsetInterface, namespace
 			allowedNamespaces = append(allowedNamespaces, namespace)
 		}
 	}
-	namespaces.Items = allowedNamespaces
-	return namespaces, nil
+	return allowedNamespaces, nil
```

Comment:
With this change, it might make sense to update the signature to accept `namespaces []corev1.Namespace` and just update callsites to call it with `.Items` on the list? Only reason is that then this can match the functional filter that you've created below (which is nicer, imo :) ). Whatever you think.